R/dataset-stream.R

```diff
@@ -51,3 +51,37 @@ appendStream <- function(ds) {
 
     return(ds)
 }
+
+
+#' Set the streaming property of a dataset
+#'
+#' Only datasets that have their streaming property set to "streaming" can
+#' have rows streamed to them. Before attempting to streaming rows (with
+#' [streamRows] for example), the dataset has to be set up to stream rows. Use
+#' `streaming(ds)` to get the streaming status, and `streaming(ds) <-
+#' "streaming"` to set the streaming status.
+#' 
+#' @param x a CrunchDataset
+#' @param value for setting only (values can be: `"no"`, `"streaming"`, or
+#' `"finished"`)
+#' 
+#' @return the streaming status
+#' @rdname streaming
+setGeneric("streaming", function(x) standardGeneric("streaming"))
+
+#' @rdname streaming
+setGeneric("streaming<-", function(x, value) standardGeneric("streaming<-"))
+
+#' @rdname streaming
+#' @export
+setMethod("streaming", "CrunchDataset", function(x) x@body$streaming)
+
+
+#' @rdname streaming
+#' @export
+setMethod("streaming<-", "CrunchDataset", function(x, value) {
```

Comment:
Is there a reason we're not using the match.call pattern here? Does that not work with setters?

Reply:
Using `match.call` would be more straightforward to do if this were a function instead of an S4 method. I'd recommend that, in fact: S4 machinery is unnecessary when we only are defining a method for one class. You could make a function and `stopifnot(is.dataset(x))`. There are some functions like this in the package.